nbgrader/api.py

```diff
@@ -1098,6 +1156,28 @@ def add_student(self, student_id, **kwargs):
         self.db.add(student)
         try:
             self.db.commit()
+            try:
+                group_name = "nbgrader-{}".format(self.course_id)
+                utils.query_jupyterhub_api(method="POST",
+                                     api_path="/groups/{name}".format(name=group_name),
+                )
+                utils.query_jupyterhub_api(method="POST",
+                                     api_path="/groups/{name}/users".format(name=group_name),
+                                     post_data = {"users":[student_id]}
+                )
+                # log.info
+                print("Student {student} added to Jupyterhub group {group_name}".format(
+                    student=student_id,
+                    group_name=group_name
+                ))
+            except utils.JupyterhubEnvironmentError as e:
+                if self.course_id: # we assume user might be using Jupyterhub but something is not working
+                    # log.error
+                    print("Student {student} NOT added to Jupyterhub group {group_name}: ".format(
```

Comment:
@jhamrick what is the best way to access logging from here? `Gradebook` doesn't have the `.log` method like the classes that derive from `NbGrader`

Reply:
My preferred way of communicating information from the `Gradebook` is to throw custom errors, and then the code using the `Gradebook` can catch those errors and provide a more useful error message to the user.